frontend/taipy-gui/src/components/Taipy/Chart.tsx

```diff
@@ -789,12 +910,77 @@ const Chart = (props: ChartProp) => {
                 }
             }
         },
-        [getRealIndex, dispatch, updateVars, propagate, props.onChange, config.traces.length, module]
+        [getRealIndex, dispatch, updateVars, propagate, props.onChange, config.traces.length, module],
     );
 
+    useEffect(() => {
+        if (!dataPl.length || !animationData) return;
+
+        const toFramesData = dataPl.map((trace) => {
+            const isYaxisAnimated = Object.hasOwn(animationData, (trace as ScatterData).name);
+            const isXaxisAnimated = (trace as ExtendedPlotData).meta?.xAxisName &&
+                Object.hasOwn(animationData, (trace as ExtendedPlotData).meta?.xAxisName as PropertyKey);
+
+            return {
+                name: (trace as ScatterData).name,
+                x: isXaxisAnimated ? animationData[(trace as ExtendedPlotData).meta?.xAxisName as keyof typeof animationData] ?? [] : (trace as ScatterData).x,
+                y: isYaxisAnimated ? animationData[(trace as ScatterData).name as keyof typeof animationData] ?? [] : (trace as ScatterData).y,
+            };
+        });
+
+        if (memoizedToFrameData !== toFramesData) {
+            setFromFrame({ ...memoizedToFrame, name: "from" });
+            setToFrame({
+                name: "to",
+                data: toFramesData,
+                traces: dataPl.map((_, idx) => idx),
+                layout: {},
+                group: "",
+                baseframe: "from",
+            });
+        }
+    }, [dataPl, animationData, memoizedToFrameData, memoizedToFrame]);
+
+    useEffect(() => {
+        if (!plotRef.current || !toFrame.data?.length) {
+            setFromFrame({
+                name: "from",
+                data: [],
+                traces: [],
+                layout: {},
+                group: "",
+                baseframe: "",
+            });
+            setToFrame({
+                name: "to",
+                data: [],
+                traces: [],
+                layout: {},
+                group: "",
+                baseframe: "",
+            });
+            return;
+        }
+
+        const plotElement = plotRef.current as unknown as PlotlyHTMLElement;
+        if (!plotElement || !plotElement.data) return;
+
+        const timer = setTimeout(() => {
+            if (plotRef.current) {
+                runAnimation().catch(console.error);
+            }
+        }, 100);
+
+        return () => {
+            if (timer) {
+                clearTimeout(timer);
+            }
+        };
+    }, [runAnimation, toFrame.data?.length]);
+
     return render ? (
         <Tooltip title={hover || ""}>
-            <Box id={id} className={`${className} ${getComponentClassName(props.children)}`} ref={plotRef}>
+            <Box id={"graph"} className={`${className} ${getComponentClassName(props.children)}`} ref={plotRef}>
```

Comment:
So where `props.id` is not used anymore in for the generated component?

Reply:
I missed that one ...